src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/payment/processor/PaymentForExistingCaseTest.java

```diff
@@ -0,0 +1,93 @@
+package uk.gov.hmcts.reform.bulkscan.payment.processor;
+
+import com.google.common.collect.ImmutableMap;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.ccd.CcdAuthenticator;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.ccd.CcdAuthenticatorFactory;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.helper.CaseSearcher;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.helper.ExceptionRecordCreator;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.helper.PaymentsMessageSender;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.model.CreatePaymentsCommand;
+import uk.gov.hmcts.reform.bulkscan.payment.processor.model.PaymentData;
+import uk.gov.hmcts.reform.ccd.client.CoreCaseDataApi;
+import uk.gov.hmcts.reform.ccd.client.model.CaseDetails;
+
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+@SpringBootTest
+@ActiveProfiles("functional")
+class PaymentForExistingCaseTest {
+
+    private static final String EXCEPTION_RECORD_AWAITING_PAYMENT_DCN_PROCESSING = "awaitingPaymentDCNProcessing";
```

Comment:
This could be `AWAITING_DNC_PROCESSING_FLAG_NAME`, to indicate what this field represents. Just a suggestion.

Reply:
ok